### PR TITLE
Improve ParamSpec bindings

### PR DESCRIPTION
**Pull request by sdroege**
* [x] Missing param spec subclasses
* [x] At least some tests

### CODE REVIEW

File: src/param_spec.rs [hunk 1]
```diff
@@ -547,3 +605,230 @@ impl ParamSpec {
...
+    StaticType + FromGlibPtrNone<*mut gobject_sys::GParamSpec> + 'static
+{
+}
+
+extern "C" {
+    pub static g_param_spec_types: *const glib_sys::GType;
+}
+
+macro_rules! define_param_spec {
+    ($typ:ident, $ffi_typ:path, $mod_name:ident, $typ_offset:expr) => {
```
Comment (GuillaumeGomez) [inline src/param_spec.rs:619]: You could rename `$ffi_typ` into `$ffi_type` and `$typ_offset` into `$type_offset`.

Comment (sdroege) [reply src/param_spec.rs:619]: Indeed, just seems inconsistent with `$typ` then :) Any suggestions for making it the same?

Comment (GuillaumeGomez) [reply src/param_spec.rs:619]: `$rust_type`?

Comment (sdroege) [reply src/param_spec.rs:619]: Thanks, better :)

**Comment by sdroege** [comment]
All but the last commit should go into a bugfix release :)

**Comment by GuillaumeGomez** [comment]
You still need to add some tests before that. ;) Otherwise no issue for the bugfix release.

**Comment by sdroege** [comment]
> You still need to add some tests before that. ;)

Tests for? The existing tests found these bugs, just had to run in valgrind :)

**Comment by GuillaumeGomez** [comment]
You listed that in your first comment here. Just quoting. :)

**Comment by sdroege** [comment]
Ah, those are in the last commit.

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
Oh indeed, went too fast and didn't see them. Looks good to me! :+1:

**Pull request merged**